tests/func-tests/node_placement_test.go

```diff
@@ -73,14 +153,16 @@ var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:sys
 				delete(expectedWorkloadsPods, "secondary-dns")
 			}
 
-			By("Listing pods in infra node")
-			pods := listPodsInNode(client, workloadsNode.Name)
+			Eventually(func(g Gomega) {
+				By("Listing pods in infra node")
+				pods := listPodsInNode(client, workloadsNode.Name)
 
-			By("Collecting nodes of pods")
-			updatePodAssignments(pods, expectedWorkloadsPods, "workload", workloadsNode.Name)
+				By("Collecting nodes of pods")
+				updatePodAssignments(pods, expectedWorkloadsPods, "workload", workloadsNode.Name)
 
-			By("Verifying that all expected workload pods exist in workload nodes")
-			Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
+				By("Verifying that all expected workload pods exist in workload nodes")
+				Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
```

Comment:
`Expect` that failed, will exit the eventually loop. Use the Gomega object to break only the current loop:
```suggestion
				g.Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
```

Reply:
You'll also nee to check the functions called from here. If they call `Expect` or `ExpectWithOffset`, you need to do the same (i.e. add new Gomega parameter to these function and use it inside).